preprocessing/BETDisk/BETTranslatorInterface.py

```diff
@@ -0,0 +1,844 @@
+
+    :param xrotorFileName: string, filepath to the Xrotor/DFDC file we want to translate into a BET disk
+    :param axisOfRotation: [x,y,z] coordinates of the rotation vector
+    :param centerOfRotation: [x,y,z] coordinates of the rotation vector
+    :param rotationDirectionRule: string, either "rightHand" or "leftHand". See https://docs.flexcompute.com/projects/flow360/en/latest/capabilities/bladeElementTheory.html#bet-input
+    :param kwargs: various other arguments see https://docs.flexcompute.com/projects/flow360/en/latest/capabilities/bladeElementTheory.html#bet-input
+    :return: dictionary that we should append to the Flow360.json file we want to run with.
+    """
+
+    diskThickness = kwargs['diskThickness']
+    gridUnit = kwargs['gridUnit']
+    chordRef = kwargs.pop('chordRef', 1.0)
+    nLoadingNodes = kwargs.pop('nLoadingNodes', 20)
+    tipGap = kwargs.pop('tipGap', 'inf')
+    initialBladeDirection = kwargs.pop('initialBladeDirection', [1, 0, 0])
+
+    if rotationDirectionRule not in ['rightHand', 'leftHand']:
+        print('Invalid rotationDirectionRule of {}. Exiting.'.format(rotationDirectionRule))
+        quit()
+    if len(axisOfRotation) != 3:
+        print('axisOfRotation must be a list of size 3. Exiting.')
+        quit()
+    if len(centerOfRotation) != 3:
+        print('centerOfRotation must be a list of size 3. Exiting')
+        quit()
+
+    xrotorDict = readXROTORFile(xrotorFileName)
+
+
+    diskJSON = {'axisOfRotation': axisOfRotation,
+                'centerOfRotation': centerOfRotation,
+                'rotationDirectionRule': rotationDirectionRule}
+
+    # xrotorInflowMach = xrotorDict['vel'] / xrotorDict['vso']
+
+    diskJSON['omega'] = xrotorDict['omegaDim'] * gridUnit / xrotorDict['vso']  # check this
+    diskJSON['numberOfBlades'] = xrotorDict['nBlades']
+    diskJSON['radius'] = xrotorDict['rad'] / gridUnit
+    diskJSON['omega'] = xrotorDict['omegaDim'] * gridUnit / xrotorDict['vso']
+    diskJSON['twists'] = generateTwists(xrotorDict, gridUnit)
+    diskJSON['chords'] = generateChords(xrotorDict, gridUnit)
+    diskJSON['MachNumbers'] = generateMachs()
+    diskJSON['alphas'] = generateAlphas()
+    diskJSON['ReynoldsNumbers'] = generateReys()
+    diskJSON['thickness'] = diskThickness
+    diskJSON['chordRef'] = chordRef
+    diskJSON['nLoadingNodes'] = nLoadingNodes
+    diskJSON['tipGap'] = tipGap
+    diskJSON['sectionalRadiuses'] = [diskJSON['radius'] * r for r in xrotorDict['rRstations']]
+    diskJSON['initialBladeDirection'] = initialBladeDirection
+    diskJSON['sectionalPolars'] = []
+
+    for secId in range(0, xrotorDict['nAeroSections']):
+        polar = getPolar(xrotorDict, diskJSON['alphas'], diskJSON['MachNumbers'], secId)
+        diskJSON['sectionalPolars'].append(polar)
+
+    return diskJSON
+
+
+########################################################################################################################
+def main():
```

Comment:
This is not an example file, the main () function is there just to test the functions above. It is the same thing as in the sampleXrotorTranslateScript.py or the dfdc one. I left it there in case we wanted a test functionality in a way that did not involve multiple files.